egs/wsj/s5/steps/nnet3/cldnn/train.sh

```diff
@@ -0,0 +1,721 @@
+#!/bin/bash
+
```

Comment:
I agree that we need a better way to deal with the commonalities in the
nnet3 training scripts.  E.g. making the config-generation a separate call
than the main network training.  But this is a big project and I don't
think it should be mixed up with the current task of checking in a CNN
recipe.

Dan


On Fri, Feb 5, 2016 at 12:31 AM, Vijayaditya Peddinti <
notifications@github.com> wrote:

> In egs/wsj/s5/steps/nnet3/cldnn/train.sh
> <https://github.com/kaldi-asr/kaldi/pull/486#discussion_r51979451>:
>
> > @@ -0,0 +1,721 @@
> > +#!/bin/bash
> > +
>
> @danpovey <https://github.com/danpovey> there is redundancy in
> steps/nnet3/
>
> */train.sh scripts as we add more architectures. Some of these scripts
> differ in nothing but the config generators. e.g. cldnn/train.sh and
> lstm/train.sh would be very similar. cnn/train.sh tdnn/train.sh would be
> very similar. I think we should look at a better way to pass the nnet3
> architecture parameters. I am thinking about passing all the make_config.py
> parameters using a config file so that steps/nnet3/*/train.sh script can
> just invoke the config generator as
>
> steps/nnet3/$model/make_configs.py $config_file
>
> Any suggestions ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/486/files#r51979451>.
>